src/components/Search.vue

```diff
@@ -0,0 +1,192 @@
+
+<script>
+import { DebounceCounter } from '../utils';
+
+const counter = new DebounceCounter();
+
+export default {
+  props: {
+    maxQuickResults: {
+      type: Number,
+      default: 6,
+    },
+    showMore: {
+      type: Boolean,
+      default: true,
+    },
+  },
+  inject: ['girderRest'],
+  data() {
+    return {
+      searchText: '',
+      searchMode: 'prefix',
+      searchTypes: ['user', 'folder', 'item'],
+      searchOptionsMenu: false,
+    };
+  },
+  computed: {
+    loading() {
+      return counter.flag;
+    },
+    quickResults() {
+      return this.searchResults.slice(0, this.maxQuickResults);
+    },
+    searchParams() {
+      return {
+        q: this.searchText,
+        mode: this.searchMode,
+        types: JSON.stringify(this.searchTypes),
+        // + 1 to determine if total results > maxQuickResults
+        limit: this.maxQuickResults + 1,
+      };
+    },
+  },
+  asyncComputed: {
+    searchResults: {
+      default: [],
+      async get() {
+        let results = [];
+        counter.inc();
+        try {
+          if (this.searchText) {
+            const { data } = await this.girderRest.get('resource/search', {
+              params: this.searchParams,
+            });
+            results = [].concat(...this.searchTypes.map(t => data[t]));
+          }
+        } catch (err) {
+          this.$emit('error', err.message || 'Unknown error during search');
+        }
+        counter.dec();
+        return results;
+      },
+    },
+  },
+  methods: {
+    formatUsername(user) {
+      return `${user.firstName} ${user.lastName} (${user.login})`;
+    },
+  },
+};
+</script>
+
+<template lang="pug">
+v-layout.girder-searchbar(row, align-center)
+  v-icon.mdi-24px(color="white") {{ $vuetify.icons.search }}
+  v-menu.grow.mx-3(
+      offset-y, content-class="girder-searchbar-menu", :open-on-click="false",
+      :value="searchText", :nudge-bottom="6")
```

Comment:
I tried out adding `transition="slide-y-transition"` and it looked pretty nice to me, should we add a transition to this menu opening? @jtomeck

Reply:
Looked fine to me, added it.  Can remove if @jtomeck feels otherwise.